app/services/import_polygons.rb

```diff
@@ -41,4 +45,48 @@ def location_categories_include?(region_name)
       location_type == :london_boroughs && DOWNCASE_BOROUGHS.include?(region_name) ||
       location_type == :cities && DOWNCASE_CITIES.include?(region_name)
   end
+
+  def get_buffers(points)
+    # ArcGIS API documentation: https://ons-inspire.esriuk.com/arcgis/sdk/rest/index.html#//02ss0000003z000000
+    buffer_distances_in_miles = [5, 10, 15, 20, 25]
+    buffered_boundaries = {}
+    buffer_distances_in_miles.each do |distance|
+      # convert 1D array, to 2D array for arcgis
+      polygon_coords = points.each_slice(2).to_a
+      response = HTTParty.get(buffer_api_endpoint(polygon_coords, convert_miles_to_metres(distance)))
+      # Buffer coordinates are stored as a 1D array
+      buffer_coords = response.dig("geometries", 0, "rings").flatten
+      buffered_boundaries[distance.to_s] = buffer_coords
+    end
+    buffered_boundaries
+  end
+
+  def buffer_api_endpoint(coords, distance)
+    geometries_param = {
+      "geometryType" => "esriGeometryPolygon",
+      "geometries" => [{ "rings" => [coords] }],
+    }
+
+    params = { "geometries" => geometries_param.to_s,
+               "inSR" => "4326",
+               "outSR" => "4326",
+               "bufferSR" => "3857",
+               "distances" => distance.to_s,
+               "unit" => "",
+               "unionResults" => "true",
+               "geodesic" => "false",
+               "f" => "json" }.to_param
+
+    api_endpoint = BUFFER_API_URL + params
+
+    if api_endpoint.length > 2000
+      # Reduce number of coordinates in order to have fewer than 2000 characters in request url
+      coords = coords.each_with_index
+      .map { |item, index| item if (index % 5).zero? }
+      .compact
```

Comment:
Can you indent `.map` and `.compact` with `.each_with_index` though, please?

Reply:
Actually, why are you using both `.each_with_index` and `.map` like that?

Reply:
This has made me realise we should test this part of the code - it won't be run by the specs because we don't give the specs incredibly long lists of points. But we still should test that we aren't effing up larger polygons. If this part of the code could be extracted to a method of its own, that would make it easier to unit test it.

Reply:
I think the easiest bit to extract to a method would be 
```ruby
coords = coords.each_with_index.map { |item, index| item if (index % 5).zero? }.compact
```
because it doesn't get involved with recursion.

Reply:
Makes for a bit more readable code at the same time - the method name will help explain what is going on

Reply:
> Actually, why are you using both .each_with_index and .map like that?

We need the index. Apparently, while there is no `map_with_index`, there is a `map.with_index`.

Reply:
I don't think map is necessary. @david-mears-dfe ?

Reply:
Can you prove it with a unit test, please?